core/metacling/src/TCling.cxx

```diff
@@ -1075,6 +1076,19 @@ inline bool TCling::TUniqueString::Append(const std::string& str)
    return notPresent;
 }
 
+void TCling::CallPedantic() {
+   fPedanticInterp->installLazyFunctionCreator(llvmLazyFunctionCreator);
+   fPedanticInterp->enableDynamicLookup();
+   fPedanticInterp->getCI()->getPreprocessorOpts().DisablePCHValidation = true;
+   fPedanticInterp->getCI()->getLangOpts().SpellChecking = false;
+   const cling::InvocationOptions& Opts = fPedanticInterp->getOptions();
+
+   cling::UserInterface Ui(*fPedanticInterp);
+   Ui.runInteractively(Opts.NoLogo);
```

Comment:
We will need ROOT's (`TApplication`'s) event loop. Running `.x tutorials/hsimple.C` in pedantic mode should show why: we need to interleave the prompt with GUI and timer event handling.